Implementation of GOE resonance sampling

Oncure opened the pull request:
This implementation was somewhat messy. I will have to revise it later.

Naww137 commented on syndat/sample_levels.py:
```diff
@@ -59,31 +59,41 @@ def wigner_PDF(x, avg_level_spacing):
-            level_spacing.append(spacing.item())
+        # Use resonance_generator from FUDGE:
+        import sys
+        sys.path.insert(0, '/Users/colefritsch/ENCORE/syndat/nuc_syndat/fudge/')
```
We need to find a better solution for user-specific path requirements. I have a similar challenge when trying to wrap around SAMMY. I will look into a way to set these as variables during the build.

Naww137 reviewed (commented):
Nice work Cole! I have a few thoughts on the overall structure though. I think the getFakeResonacneSet function outputs both levels and widths. If this is the case, I would like to avoid calling it twice, though I appreciate you trying to keep the inputs/outputs here the same! Perhaps we can take this one level up such that the function particle_pair.sample_resonance_ladder() inputs/output the same format. Within that function we can have some logic that either calls these functions (for if the user does not have access to fudge) or calls a separate function that houses what you have written.

Oncure commented:
Hey Noah. My work load is picking up again and I will not have much time to work on research. It is looking like I won't get a break until winter. Do you think I can finalize this over winter break? Sorry for the delay.

I have reread your comment more carefully, and noticed a misunderstanding. "getFakeResonanceSet" can give widths, but I am not generating widths with it, just energies. Later, I can implement it more thoroughly so that I can put in widths, but I need time to learn about the "xData" tool they are using.

I believe you have an unused redundant function, "sample_resonance_levels", which is effectively the same as the "sample_RRR_levels". I have implemented the call in both. I was going to let you know of the redundancies when I sent a commit message, but I forgot.

Oncure committed:
Cleaning, adding comments, and editting Fudge implementation.

Oncure committed:
Example notebook commit.

Oncure committed:
Merge branch 'main' of https://github.com/Naww137/nuc_syndat

Oncure committed:
Bug fixing for Fudge implementation.

Oncure committed:
More fudge implementation bugfixing

Oncure committed:
Final commit.

Oncure committed:
Merge branch 'main' of https://github.com/Naww137/nuc_syndat

Oncure committed:
Updating for functional and json setting selection.

Oncure committed:
Merge branch 'main' of https://github.com/Naww136/nuc_syndat

Oncure commented:
I updated Syndat to now allow both the functional and json settings. I am not great with git, so there may be some errors I have not noticed that have arose from merging. The functional and json selection function slightly differently since we had our own slightly different FUDGE and SAMMY option implementation, but it should work for either case.

Oncure committed:
Debugged particle_pair file.

Oncure committed:
Restored the GenerateData_UserExample.ipynb file.

Oncure committed:
Merge branch 'main' of https://github.com/Naww137/nuc_syndat

Oncure committed:
Updating MMDA

Oncure committed:
Merge branch 'main' of https://github.com/Naww137/nuc_syndat

Oncure committed:
Merge branch 'main' of https://github.com/Naww137/nuc_syndat

Oncure committed:
Removed a comment

Oncure committed:
Removing FUDGE from git.

Oncure commented:
Okay, my implementation is up to date, but I will also make a simpler version if you think that would be better.

Oncure committed:
Updated particle_pair.py and its fudge sampling. error checks and messages were added.

Oncure commented:
I updated the JSON input implementation as a factory class method so that it does not obstruct the original parameter inputs. I separated the FUDGE call as a separate function for readability as you suggested. I also added extra error messages, including a "possible_options" dictionary that reflects the "default_options" dictionary.
